static/skin/index.js

```diff
@@ -207,34 +207,42 @@
         const zimUrl = params.get('as'); // as= is fallback URL
         // download metalink to build list of mirrored URLs
         let mirrorUrls = [];
-        try {
-            const metalink = await fetch(`${zimUrl}.meta4`).then(response => {
-                return response.ok ? response.text() : '';
-            }).catch((_error) => '');
-            if (metalink) {
+
+        const metalink = await fetch(`${zimUrl}.meta4`).then(response => {
+            return response.ok ? response.text() : '';
+        }).catch((_error) => '');
+        if (metalink) {
+            try {
                 const parser = new DOMParser();
                 const doc = parser.parseFromString(metalink, "application/xml");
                 doc.querySelectorAll("url").forEach((node) => {
                     if (node.hasAttribute("priority")) { // ensures its a mirror link
                         mirrorUrls.push(node.innerHTML);
                     }
                 });
+            } catch (err) {
+                // not a big deal, magnet will only contain primary URL
+                console.debug(`Failed to parse mirror links for ${zimUrl}`);
             }
-        } catch (err) {
-            // not a big deal, magnet will only contain primary URL
-            console.debug(`Failed to retrieve mirror links for ${zimUrl}`);
         }
 
         // set webseed (ws=) URL to primary download URL (redirects to mirror)
-        params.set('ws', zimUrl);
+        params.set('ws', encodeURIComponent(zimUrl));
         // if we got metalink mirror URLs, append them all
         if (mirrorUrls) {
             mirrorUrls.forEach((url) => {
-                params.append('ws', url);
+                params.append('ws', encodeURIComponent(url));
             });
         }
 
-        params.set('xs', `${zimUrl}.torrent`);  // adding xs= to point to torrent URL
+        params.set('as', encodeURIComponent(zimUrl));
+        params.set('dn', encodeURIComponent(params.get('dn')));
+        params.set('xs', encodeURIComponent(`${zimUrl}.torrent`));  // adding xs= to point to torrent URL
+        const trackers = params.getAll('tr');
+        params.delete('tr');
+        trackers.forEach((tracker) => {
+            params.append('tr', encodeURIComponent(tracker));
+        });
 
         magnetLink = 'magnet:?';
         // we don't want params.ToString()'s urlencoding
```

Comment:
This comment is now misleading

Reply:
No it's not. We still don't want [URLSearchParams.toString](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/toString) because it URL encodes everything like the `:` in URN and all. Result is thus very different from what we produce here and most importantly it's rejected by torrent clients.

Reply:
`encodeURIComponent()` encodes everything, too. Did you mean to use `encodeURI()`?

Reply:
No, precisely because we're applying it to URLs only.